singularity-eos/eos/singularity_eos.cpp

```diff
@@ -35,15 +35,30 @@ int init_sg_eos(const int nmat, EOS *&eos) {
   return 0;
 }
 
-#define SGAPPLYMOD(A)                                                                    \
-  EOSBuilder::applyShiftAndScale(A, enabled[0] == 1, enabled[1] == 1, vals[0], vals[1])
+// apply everything but ramp in order to possibly calculate the 
+// SAP ramp parameters from p-alhpa ramp parameters
+#define SGAPPLYMODSIMPLE(A) \
+  EOSBuilder::applyShiftAndScale(A, enabled[0] == 1, enabled[1] == 1, \
+				 vals[0], vals[1])
 
-constexpr const int def_en[2] = {0, 0};
-constexpr const double def_v[2] = {0.0, 0.0};
+#define SGAPPLYMOD(A) \
+  EOSBuilder::applyShiftAndScaleAndSAPRamp(A, enabled[0] == 1, enabled[1] == 1, \
+                                           enabled[2] == 1 || enabled[3] == 1, \
+					   vals[0], vals[1], vals[2], vals[3], \
+					   vals[4], vals[5])
+
+constexpr const int def_en[4] = {0, 0, 0, 0};
+constexpr const double def_v[6] = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
 
 int init_sg_IdealGas(const int matindex, EOS *eos, const double gm1, const double Cv,
-                     int const *const enabled, double const *const vals) {
+                     int const *const enabled, double const *const vals_i) {
   assert(matindex >= 0);
+  Real vals[6] = {vals_i[0], vals_i[1], vals_i[2], vals_i[3], vals_i[4], vals_i[5]};
```

Comment:
This is how I'm currently handling the multiple ways to initialize a ramp that covers standard initialization and initialization from p-alpha model parameters. If this looks ok, I'll do this for all of the other Eos's. Would like some eyes maybe @Yurlungur and @jhp-lanl

Reply:
This seems fine to me. The names are a little vague, though. What is `vals` vs `vals_i`?

Reply:
I think letting them be non-const is the cleaner and clearer design

Reply:
@Yurlungur this has now been addressed.